patina_dxe_core/src/dxe_services.rs

```diff
@@ -370,130 +367,138 @@ extern "efiapi" fn get_io_space_map(
     }
 }
 
-extern "efiapi" fn dispatch() -> efi::Status {
-    match core_dispatcher() {
-        Err(err) => err.into(),
-        Ok(()) => efi::Status::SUCCESS,
-    }
-}
+impl<P: PlatformInfo> Core<P> {
+    /// Initializes and installs the DXE Services table into the provided system table.
+    pub(crate) fn install_dxe_services_table(&self, system_table: &mut EfiSystemTable) {
+        let mut dxe_system_table = dxe_services::DxeServicesTable {
+            header: efi::TableHeader {
+                signature: efi::BOOT_SERVICES_SIGNATURE,
+                revision: efi::BOOT_SERVICES_REVISION,
+                header_size: mem::size_of::<dxe_services::DxeServicesTable>() as u32,
+                crc32: 0,
+                reserved: 0,
+            },
+            add_memory_space,
+            allocate_memory_space,
+            free_memory_space,
+            remove_memory_space,
+            get_memory_space_descriptor,
+            set_memory_space_attributes,
+            get_memory_space_map,
+            add_io_space,
+            allocate_io_space,
+            free_io_space,
+            remove_io_space,
+            get_io_space_descriptor,
+            get_io_space_map,
+            dispatch: Self::dispatch_efiapi,
+            schedule: Self::schedule_efiapi,
+            trust: Self::trust_efiapi,
+            process_firmware_volume: Self::process_firmware_volume_efiapi,
+            set_memory_space_capabilities,
+        };
+        let dxe_system_table_ptr = &dxe_system_table as *const dxe_services::DxeServicesTable;
+        let crc32 = unsafe {
+            crc32fast::hash(from_raw_parts(
+                dxe_system_table_ptr as *const u8,
+                mem::size_of::<dxe_services::DxeServicesTable>(),
+            ))
+        };
+        dxe_system_table.header.crc32 = crc32;
+
+        let dxe_system_table = Box::new_in(dxe_system_table, &EFI_RUNTIME_SERVICES_DATA_ALLOCATOR);
```

Comment:
Can this be Boot Services Data? I'm not sure it needs to be runtime. I think it is in the global data section of DxeMain in the EDK2 core which would be BSD.

Reply:
Switched to a normal `Box` and `Box::into_raw`, as the global allocator is the BOOT_SERVICES_DATA_ALLOCATOR